psydac/api/postprocessing.py

```diff
@@ -814,170 +961,522 @@ def export_to_vtk(self, filename_pattern, grid, npts_per_cell=None, snapshots='n
+                                top_left_front, top_right_front, bot_right_front, bot_left_front,
+                                top_left_back, top_right_back, bot_right_back, bot_left_back
+                            ]
+
+                            cellID += 1
+
+        else:
+            raise NotImplementedError("Not Supported Yet")
+
+        return connectivity, offsets, celltypes, cellshape
+
+
+def _augment_space_degree_dict(ldim, sequence='DR'):
+    """
+    With the 'DR' sequence in 3D, all multiplicies are [r1, r2, r3] and we have
+     'H1'   : degree = [p1, p2, p3]
+     'Hcurl': degree = [[p1-1, p2, p3], [p1, p2-1, p3], [p1, p2, p3-1]]
+     'Hdiv' : degree = [[p1, p2-1, p3-1], [p1-1, p2, p3-1], [p1-1, p2-1, p3]]
+     'L2'   : degree = [p1-1, p2-1, p3-1]
+
+    With the 'TH' sequence in 2D we have:
+     'H1' : degree = [[p1, p2], [p1, p2]], multiplicity = [[r1, r2], [r1, r2]]
+     'L2' : degree = [p1-1, p2-1], multiplicity = [r1-1, r2-1]
+
+    With the 'RT' sequence in 2D we have:
+    'H1' : degree = [[p1, p2-1], [p1-1, p2]], multiplicity = [[r1,r2], [r1,r2]]
+    'L2' : degree = [p1-1, p2-1], multiplicity = [r1, r2]
+
+    With the 'N' sequence in 2D we have:
+    'H1' : degree = [[p1, p2], [p1, p2]], multiplicity = [[r1,r2+1], [r1+1,r2]]
+    'L2' : degree = [p1-1, p2-1], multiplicity = [r1, r2]
+    """
+    assert ldim in [2, 3]
+    
+    if sequence == 'DR':
+        def f_h1(degree, multiplicity):
+            if isinstance(degree[0], (list, tuple, np.ndarray)):
+                return degree[0], multiplicity
+            else:
+                return degree, multiplicity
+    
+        def f_l2(degree, multiplicity):
+            if isinstance(degree[0], (list, tuple, np.ndarray)):
+                return np.asarray(degree[0]) + 1, multiplicity
+            else:
+                return np.asarray(degree) + 1, multiplicity
+
+        if ldim == 2:
+            def f_hdiv(degree, multiplicity):
+                degree = np.asarray(degree[0])
+                degree += np.array([0, 1])
+                return degree, multiplicity
+
+            def f_hcurl(degree, multiplicity):
+                degree = np.asarray(degree[0])
+                degree += np.array([1, 0])
+                return degree, multiplicity
+
+        else:
+            def f_hdiv(degree, multiplicity):
+                degree = np.asarray(degree[0])
+                degree += np.array([0, 1, 1])
+                return degree, multiplicity
+
+            def f_hcurl(degree, multiplicity):
+                degree = np.asarray(degree[0])
+                degree += np.array([1, 0, 0])
+                return degree, multiplicity 
+
+        kind_dict = {
+            'h1': f_h1,
+            'hcurl': f_hcurl,
+            'hdiv': f_hdiv,
+            'l2': f_l2,
+            'undefined': f_h1, 
+        }
+        return kind_dict
+    else:
+        raise NotImplementedError("Only DR sequence is implemented")
```

Comment:
Missing newline at end of file

Reply:
Added in [b8b1cea](https://github.com/pyccel/psydac/pull/214/commits/b8b1cea2241f1cb78baccab3fd96192b277d57f5)